AEPMessaging/Sources/Inbound.swift

```diff
@@ -50,54 +50,46 @@ public class Inbound: NSObject, Codable {
-        let values = try decoder.container(keyedBy: CodingKeys.self)
-
-        uniqueId = try values.decode(String.self, forKey: .id)
-        if let format = try? values.decode(String.self, forKey: .inboundType) {
-            inboundType = InboundType(from: format)
-        } else {
-            // TODO: - use regex to deduce inbound content format from the content string
-            inboundType = .unknown
-        }
-        contentType = try values.decode(String.self, forKey: .contentType)
-        publishedDate = try values.decode(Int.self, forKey: .publishedDate)
-        expiryDate = try values.decode(Int.self, forKey: .expiryDate)
-        let codableMeta = try? values.decode([String: AnyCodable].self, forKey: .meta)
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+
+        uniqueId = try container.decode(String.self, forKey: .id)
+        inboundType = try InboundType(from: container.decode(String.self, forKey: .type))
+        contentType = try container.decode(String.self, forKey: .contentType)
+        publishedDate = try container.decode(Int.self, forKey: .publishedDate)
+        expiryDate = try container.decode(Int.self, forKey: .expiryDate)
+        let codableMeta = try? container.decode([String: AnyCodable].self, forKey: .meta)
         meta = codableMeta?.mapValues {
             guard let value = $0.value else {
                 return ""
             }
             return value
         }
 
-        let codableContent = try values.decode(AnyCodable.self, forKey: .content)
-        if contentType == "application/json" {
-            if
-                let jsonData = codableContent.dictionaryValue,
+        let codableContent = try container.decode(AnyCodable.self, forKey: .content)
+        if let inboundContent = codableContent.stringValue {
+            content = inboundContent
+        } else if let jsonData = codableContent.dictionaryValue {
+            guard
                 let encodedData = try? JSONSerialization.data(withJSONObject: jsonData),
                 let inboundContent = String(data: encodedData, encoding: .utf8)
-            {
-                content = inboundContent
-                return
+            else {
+                throw DecodingError.dataCorrupted(DecodingError.Context(codingPath: decoder.codingPath,
+                                                                        debugDescription: "Inbound content dictionary is invalid."))
             }
+            content = inboundContent
         } else {
-            if let inboundContent = codableContent.stringValue {
-                content = inboundContent
-                return
-            }
+            throw DecodingError.typeMismatch(Inbound.self,
+                                             DecodingError.Context(codingPath: decoder.codingPath,
+                                                                   debugDescription: "Inbound content is not of an expected type."))
```

Comment:
what do you think about making `content` an `Any` so we can have it represented as a String or Dictionary based on `contentType`?

Reply:
We could, however decoding allows us to validate the content. And store it as valid string (or JSON string) which we can further decode as one of the supported internal types e.g. Feed, native Alert etc with ease.